java-manta-client-unshaded/src/main/java/com/joyent/manta/client/MantaClient.java

```diff
@@ -186,6 +189,13 @@
      */
     private final MantaMBeanSupervisor beanSupervisor;
 
+    /**
+     * ForkJoinPool used specifically for find() operations because we want
+     * to make sure that the number of concurrent threads will not exceed
+     * the maximum number of available connections.
+     */
+    private final ForkJoinPool findForkJoinPool;
```

Comment:
Per some of the discussion in #316, should this class be package private?

Reply:
I reread #316, but I'm not understanding why we would want to change the scope from private to package private. Wouldn't that _increase_ the surface area for compatibility problems?